api/lib/searchOperators.js

```diff
@@ -1,4 +1,171 @@
+      // eslint-disable-next-line no-control-regex
+      const seperateAtPlusorMinus = /[+-]?[\x00-\x2A\x2C\x2A\x2E-\x7F]+/g;
+      const matches = modifiedSearchQuery.match(seperateAtPlusorMinus);
+
+      const conditions = [];
+      const parameters = [];
+
+      matches.forEach(match => {
+        // !match.includes('+') && !match.includes('-') means this is the very first part of the query
+        // which is implicitly a plus (e.g. Awip+inrMjnu) means (e.g. +Awip+inrMjnu)
+        if (match.includes('+') || (!match.includes('+') && !match.includes('-'))) {
+          // remove + if it exists
+          let modifiedMatch = match.replace(/\++/g, '');
+          conditions.push('v.Gurmukhi LIKE BINARY ?');
+
+          // as it stands, theres really no difference between '*', "", '', etc.
+          // so the following queries give the same results
+          // "Awip" + "inrMjnu" + "Awpy" vs *Awip* + *inrMjnu* + *Awpy* vs Awip + inrMjnu + Awpy
+          if (match.includes('*')) {
+            modifiedMatch = modifiedMatch.replace(/\*+/g, constantsObj.AsteriskMariadbTranslation);
+          } else if (match.includes('"') || match.includes("'")) {
+            modifiedMatch = modifiedMatch.replace(/"+/g, '');
+            modifiedMatch = modifiedMatch.replace(/'+/g, '');
+          }
+
+          // remove spaces
+          modifiedMatch = `%${modifiedMatch.replace(/\s+/g, '')}%`;
+          parameters.push(modifiedMatch);
+        } else if (match.includes('-')) {
+          // remove - if it exists
+          let modifiedMatch = match.replace(/-+/g, '');
+          conditions.push('v.Gurmukhi NOT LIKE BINARY ?');
+
+          if (match.includes('*')) {
+            modifiedMatch = modifiedMatch.replace(/\*+/g, constantsObj.AsteriskMariadbTranslation);
+          } else if (match.includes('"') || match.includes("'")) {
+            modifiedMatch = modifiedMatch.replace(/"+/g, '');
+            modifiedMatch = modifiedMatch.replace(/'+/g, '');
+          }
+
+          modifiedMatch = `%${modifiedMatch.replace(/\s+/g, '')}%`;
+          parameters.push(modifiedMatch);
+        }
+      });
+
+      if (matches.length > 0) {
+        return {
+          condition: conditions.join(' AND '),
+          parameters,
+        };
+      }
+
+      // in the case they only have an asterisk or quotes, just clean up the operators
+      modifiedSearchQuery = modifiedSearchQuery.replace(
+        /\*+/g,
+        constantsObj.AsteriskMariadbTranslation,
+      );
+      modifiedSearchQuery = modifiedSearchQuery.replace(/"+/g, '');
+      modifiedSearchQuery = modifiedSearchQuery.replace(/'+/g, '');
+      return {
+        condition: 'v.Gurmukhi LIKE BINARY ?',
+        parameters: [modifiedSearchQuery],
+      };
+    }
+    return {
+      condition: 'v.Gurmukhi LIKE BINARY ?',
+      parameters: [`%${modifiedSearchQuery}%`],
+    };
+  },
+  fullWordEnglishToQuery: searchQuery => {
```

Comment:
tbh this function is very similar to the above `fullWordGurmukhiToQuery`. I think we can definitely abstract them out to cut down on the amount of code as a future improvement.

Reply:
agree